hooks/checkout

```diff
@@ -90,6 +97,8 @@ checkout() {
   git reset --hard
   git clean -ffxdq
 
+  trap 'buildkite-agent artifact upload "${git_log}" && rm -rf "${git_log}"' EXIT
```

Comment:
I wasn't sure the best way to do this, I was messing around with a function that changes a trap on the fly, but that was taking a fair bit of time.

I ended up just combining it all into one trap and  making the git log directory a global dir.  Do you think this is OK @uri-canva?

Reply:
Yeah, that's ok.